src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java

```diff
@@ -60,15 +58,16 @@ private static double convertTo5dp(double num) {
             }
 
             // when the answer is within range but not one of the possible values
-            double interval = details.getAnswer() - minScale;
-            double remainder = convertTo5dp(interval - Math.floor(interval / step) * step);
-            boolean isAnswerNotAPossibleValueWithinRange = remainder != 0.0 && !isAnswerOutOfRange;
+            BigDecimal answer = BigDecimal.valueOf(details.getAnswer());
+            BigDecimal stepBd = BigDecimal.valueOf(step);
+            BigDecimal remainder = answer.remainder(stepBd);
```

Comment:
The logic is wrong here. The remainder should be taken over `answer - minScale`, not simply `answer`. This logic will return no error with setting of minscale = 6, maxscale = 10, step = 0.7, answer = 7, while it should have returned error as the nearest answers are 6.7 and 7.4.

Reply:
oh yes, your right - I have updated this